.ci/on-schedule.sh

```diff
@@ -107,6 +109,46 @@ function collect_aur_timestamps() {
     http_proxy="$CI_AUR_PROXY" https_proxy="$CI_AUR_PROXY" UTIL_FETCH_AUR_TIMESTAMPS collect_aur_timestamps_output "${AUR_PACKAGES[*]}"
 }
 
+function collect_changed_libs() {
+    for repo in "${CI_ARCH_REPOS[@]}"; do
+        UTIL_PARSE_DATABASE "$repo" "${CI_ARCH_MIRROR}/$repo/os/x86_64/${repo}.db"
+    done
+    UTIL_PARSE_DATABASE "${REPO_NAME}" "${CI_CAUR_MIRROR}/${REPO_NAME}/x86_64/${REPO_NAME}.db"
+
+    diff .ci/lib.state .ci/lib.state.new | grep -E '^< ' | cut -d' ' -f2- >.ci/lib.state.diff
+    while IFS= read -r line; do
+        IFS=':' read -r -a pkg <<<"$line"
+        CHANGED_LIBS+=("${pkg[0]}")
+    done <.ci/lib.state.diff
+}
+
+function update-lib-bump() {
+    local bump=0
+    local -n CONFIG=${1:-VARIABLES}
+
+
+    if [ ! -v CONFIG[CI_REBUILD_TRIGGERS] ]; then
+        return 0
+    fi
+
+    # Split the string on : into an array
+    IFS=':' read -r -a LIBS <<<"${CONFIG[CI_REBUILD_TRIGGERS]}"
+    for library in "${LIBS[@]}"; do
+        if [[ ${CHANGED_LIBS[*]} =~ $library ]]; then
+            bump=1
+        fi
+    done
```

Comment:
Instead of this kind of lookup, you should be using an associative array to allow for `-v` based lookups like:
`[[ -v CONFIG["${library}"] ]]`

That would make this function quite a bit smaller and performance would be much improved as well.

Reply:
I'm not sure whether I got the quite a bit smaller part right. But lets see once I push it.